python/ml4ir/base/io/file_io.py

```diff
@@ -45,6 +45,17 @@ def make_directory(dir_path: str, clear_dir: bool = False, log=None) -> str:
     return dir_path
 
 
+def read_list(infile: str):
+    """Read a text file and convert entries into a list of strings by splitting on newline"""
+    txt = ""
+    if infile.startswith("hdfs"):
+        raise NotImplementedError
+    else:
+        txt = open(infile, "r").read()
```

Comment:
same as in another part of the code. You don't need the else. Can be simplified.

Reply:
Unfortunately, here I do need it. The plan is to add the `if` section soon(as I work on enabling deepsea support for ml4ir). I didn't want to implement it as part of the other stories is all.